problem_builder/swipe.py

```diff
@@ -41,53 +43,52 @@ def _(text):
 
 
 # Classes ###########################################################
-class SwipeBlock(SubmittingXBlockMixin, StudentViewUserStateMixin, QuestionnaireAbstractBlock):
+
+@XBlock.needs("i18n")
+class SwipeBlock(
+    QuestionMixin, StudioEditableXBlockMixin, StudentViewUserStateMixin, XBlockWithPreviewMixin, XBlock,
+):
     """
     An XBlock used to ask binary-choice questions with a swiping interface
     """
     CATEGORY = 'pb-swipe'
     STUDIO_LABEL = _(u"Swipeable Binary Choice Question")
-    USER_STATE_FIELDS = ['num_attempts', 'student_choice']
-
-    message = String(
-        display_name=_("Message"),
-        help=_(
-            "General feedback provided when submitting. "
-            "(This is not shown if there is a more specific feedback tip for the choice selected by the learner.)"
-        ),
-        scope=Scope.content,
-        default=""
-    )
+    USER_STATE_FIELDS = ['student_choice']
 
-    student_choice = String(
-        # {Last input submitted by the student
+    text = String(
+        display_name=_("Text"),
+        help=_("Text to display on this card. The student must determine if this statement is true or false."),
+        scope=Scope.content,
         default="",
-        scope=Scope.user_state,
+        multiline_editor=True,
     )
 
-    correct_choice = String(
-        display_name=_("Correct Choice"),
-        help=_("Specify the value that students may select for this question to be considered correct."),
-        scope=Scope.content,
-        values_provider=QuestionnaireAbstractBlock.choice_values_provider,
-    )
     img_url = String(
         display_name=_("Image"),
-        help=_(
-            "Specify the URL of an image associated with this question."
-        ),
+        help=_("Specify the URL of an image associated with this question."),
         scope=Scope.content,
         default=""
     )
-    editable_fields = QuestionnaireAbstractBlock.editable_fields + ('message', 'correct_choice', 'img_url',)
+
+    correct = Boolean(
+        display_name=_("Correct Choice"),
+        help=_("Specifies whether the card is correct."),
+        scope=Scope.content,
+    )
+
+    student_choice = String(
```

Comment:
This should be a `Boolean` field too. (Causes errors otherwise, when storing a boolean).

Reply:
Woops, that's right.